components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -113,6 +113,7 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		workspace.Status.Conditions = []metav1.Condition{}
 	}
 
+	log = log.WithValues("owi", workspace.OWI())
```

Comment:
this would simplify it a bit, by updating the ctx with the logger that includes OWI values, any methods called by `Reconcile` can just call `log.FromContext(ctx)` to also get the owi values (so you can remove the `.WithValues("owi", ...` in those places)

```suggestion
	log = log.WithValues("owi", workspace.OWI())
	ctx = logr.NewContext(ctx, log)
```

Reply:
Oh nice! On it.

Reply:
Fixed and tested in https://cloudlogging.app.goo.gl/inS4j5o2G7PJd8U78